src/drivers/mct/main/cime_comp_mod.F90

```diff
@@ -2432,21 +2458,92 @@ subroutine cime_run()
     call seq_timemgr_EClockGetData( EClock_d, curr_ymd=ymd, curr_tod=tod)
 #ifndef CPL_BYPASS
     ! Report on memory usage
+    call shr_mem_getusage(msize,mrss)
+
+    allocate( msizeOnTask(0:npes_GLOID-1), mrssOnTask(0:npes_GLOID-1), stat=ierr)
```

Comment:
I'm not sure this is going to be fully scalable and suggest that there be a way to turn it off for large cases.